.travis.yml

```diff
@@ -0,0 +1,117 @@
+matrix:
+  include:
+    - name: "Build C++/Java and Upload to S3 on AL2012"
+      os: linux
+      services: docker
+      dist: trusty
+      before_install:
+        - echo "################################################################################"
+        - echo "Installing Dependencies for Linux"
+        - echo "################################################################################"
+        - sudo apt-get update
+        - sudo apt-get install python3
+        - sudo apt-get -y install python3-pip
+        - pip3 --version
+        - pip3 install --user awscli
+        - mkdir -p ~/.aws
+        - |
+          cat > ~/.aws/credentials << EOF
+          [default]
+          aws_access_key_id = $AWS_ACCESS_KEY_ID
+          aws_secret_access_key = $AWS_SECRET_ACCESS_KEY
+        - cat <<'EOF' >> ~/.aws/credentials
+        - echo "################################################################################"
+        - echo "SUCCESS Installed Dependencies for Linux"
+        - echo "################################################################################"
+      script:
+        - echo "################################################################################"
+        - echo "Starting build phase for Linux"
+        - echo "################################################################################"
+        - $(aws ecr get-login --no-include-email --region us-west-2)
+        - echo "Getting Docker image for AL2012 build of KPL..."
+        - docker pull 056543101242.dkr.ecr.us-west-2.amazonaws.com/kinesis-producer-library-builder:latest
+        - ls
+        - echo "Building KPL in current directory in the Container..."
+        - docker run --memory 9g -v $(pwd):/kpl -it 056543101242.dkr.ecr.us-west-2.amazonaws.com/kinesis-producer-library-builder:latest /bin/bash -c "export AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY && export AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID && cd /kpl && ls && ./bootstrap.sh"
+        - cd java/amazon-kinesis-producer/src/main/resources/amazon-kinesis-producer-native-binaries/$TRAVIS_OS_NAME
+        - ls
+        - sudo zip kinesis-producer.zip kinesis_producer
+        - echo "Uploading build artifacts for KPL built on Linux..."
+        - aws s3 cp kinesis-producer.zip s3://kpl-build-kinesis-internal/$KPL_VERSION/$TRAVIS_OS_NAME/kinesis-producer.zip
```

Comment:
Would eventual consistency be a concern for multiple versions for minor bug fixes? Not sure ho frequently KPL_VERSION is updated.

Reply:
The idea is for this script to be executed as needed during release cycles. Eventual  consistency won't matter, since the build process is approximately 40 minutes.